FindTown/FindTownUI/Sources/FindTownUI/FindTownTextField.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  File.swift
+//  
+//
+//  Created by 장선영 on 2022/12/21.
+//
+
+import Foundation
+import UIKit
+
+/// FindTownTF의 status 타입 정의
+public enum FindTownTextFieldStatus {
+    case focused   /// 작성 중
+    case unfocused /// 작성 중 아님
+    case error     /// 텍스트가 조건을 만족하지 않음
+}
+
+/// FindTown 프로젝트에서 사용되는 Custom TextField
+public class FindTownTextField: UITextField {
+    
+    /// TextField의 좌, 우 inset
+    var insetX: CGFloat = 16
+    /// TextField의 status중 error 케이스 사용 여부
+    var isErrorCaseIncluded = false
+    
+    /// FindTownTextFieldStatus에 따라 TextField layer의 색상 변경
+    public var status: FindTownTextFieldStatus = .unfocused {
+        didSet {
+            switch status {
+            case .unfocused:
+                self.layer.borderColor = FindTownColor.grey4.color.cgColor
+            case .error:
+                self.layer.borderColor = FindTownColor.semantic.color.cgColor
+            case .focused:
+                self.layer.borderColor = FindTownColor.grey6.color.cgColor
+            }
+        }
+    }
```

Comment:
혹시 에러처리를 테스트해보셨을까요? 전 상태가 바로 반영이 되지 않아서..🥲 layoutIfNeeded를 추가했을때 반영이 되었어요!

```suggestion
    public var status: FindTownTextFieldStatus = .unfocused {
        didSet {
            switch status {
            case .unfocused:
                self.layer.borderColor = FindTownColor.grey4.color.cgColor
            case .error:
                self.layer.borderColor = FindTownColor.semantic.color.cgColor
            case .focused:
                self.layer.borderColor = FindTownColor.grey6.color.cgColor
            }
            self.layoutIfNeeded()
        }
    }
```